rename artifactId

unfortunately, `maven-dependency-duplicator-plugin` is an illegal name:

> [ERROR] 
> 
> Artifact Ids of the format maven-___-plugin are reserved for 
> plugins in the Group Id org.apache.maven.plugins
> Please change your artifactId to the format ___-maven-plugin
> In the future this error will break the build.

--- a/README.md
+++ b/README.md
@@ -1,13 +1,13 @@
-![Java CI with Maven](https://github.com/mickroll/maven-dependency-duplicator-plugin/workflows/Java%20CI%20with%20Maven/badge.svg)
-![Maven Package](https://github.com/mickroll/maven-dependency-duplicator-plugin/workflows/Maven%20Package/badge.svg)
+![Java CI with Maven](https://github.com/mickroll/dependency-duplicator-plugin/workflows/Java%20CI%20with%20Maven/badge.svg)
+![Maven Package](https://github.com/mickroll/dependency-duplicator-plugin/workflows/Maven%20Package/badge.svg)
 
-# maven-dependency-duplicator-plugin
+# dependency-duplicator-plugin
 
 This Maven plugin duplicates existing dependencies, applying configured changes. Each project is treated separate, at the beginning of the reactor build.
 
 ## Usage
 
-Modifying your maven build to include the maven-dependency-duplicator-plugin is done in two steps, by registering it as a build extension and adding configuration properties.
+Modifying your maven build to include the dependency-duplicator-plugin is done in two steps, by registering it as a build extension and adding configuration properties.
 
 ### Build extension
 
@@ -17,17 +17,17 @@ Add the plugin to the maven build by registering it as build extension as follow
         <plugins>
             <plugin>
                 <groupId>com.github.madprogger</groupId>
-                <artifactId>maven-dependency-duplicator-plugin</artifactId>
+                <artifactId>dependency-duplicator-plugin</artifactId>
                 <!-- hint: don't forget to define the following property with current plugin version -->
-                <version>${maven-dependency-duplicator-plugin.version}</version>
+                <version>${dependency-duplicator-plugin.version}</version>
                 <extensions>true</extensions>
             </plugin>
         </plugins>
     </build>
 
-Now it is run in any build in the project, thus being able to modify the dependency tree. The plugin itself is provided via maven central: https://repo1.maven.org/maven2/com/github/madprogger/maven-dependency-duplicator-plugin/
+Now it is run in any build in the project, thus being able to modify the dependency tree. The plugin itself is provided via maven central: https://repo1.maven.org/maven2/com/github/madprogger/dependency-duplicator-plugin/
 
-Hint: any subsequent build extension will also see the modified dependency tree. So if you use multiple extensions, be sure to configure the maven-dependency-duplicator-plugin as the first extension.
+Hint: any subsequent build extension will also see the modified dependency tree. So if you use multiple extensions, be sure to configure the dependency-duplicator-plugin as the first extension.
 
 ### Configuration
   
@@ -60,7 +60,7 @@ The configuration is read separately for each project, so different configuratio
  
 ## Example
 
-If the following configuration is defined in the root pom.xml of a multi module maven project that uses the maven-dependency-duplicator-plugin build extension:
+If the following configuration is defined in the root pom.xml of a multi module maven project that uses the dependency-duplicator-plugin build extension:
 
     <configuration>
         <duplications>

--- a/plugin-integrationtest/pom.xml
+++ b/plugin-integrationtest/pom.xml
@@ -4,7 +4,7 @@
     xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
     <modelVersion>4.0.0</modelVersion>
     <groupId>com.github.madprogger</groupId>
-    <artifactId>maven-dependency-duplicator-plugin-integrationtest</artifactId>
+    <artifactId>dependency-duplicator-plugin-integrationtest</artifactId>
     <version>0.3.0-SNAPSHOT</version>
 
     <name>dependency duplicator build plugin - integrationtest</name>

--- a/plugin-integrationtest/src/test/maven-projects/multimodule/pom.xml
+++ b/plugin-integrationtest/src/test/maven-projects/multimodule/pom.xml
@@ -39,7 +39,7 @@
         <plugins>
             <plugin>
                 <groupId>com.github.madprogger</groupId>
-                <artifactId>maven-dependency-duplicator-plugin</artifactId>
+                <artifactId>dependency-duplicator-plugin</artifactId>
                 <version>${ddpVersion}</version>
                 <extensions>true</extensions>
                 <configuration>

--- a/plugin-integrationtest/src/test/maven-projects/simple/pom.xml
+++ b/plugin-integrationtest/src/test/maven-projects/simple/pom.xml
@@ -18,7 +18,7 @@
         <plugins>
             <plugin>
                 <groupId>com.github.madprogger</groupId>
-                <artifactId>maven-dependency-duplicator-plugin</artifactId>
+                <artifactId>dependency-duplicator-plugin</artifactId>
                 <version>${ddpVersion}</version>
                 <extensions>true</extensions>
             </plugin>

--- a/plugin/pom.xml
+++ b/plugin/pom.xml
@@ -4,19 +4,19 @@
     xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
     <modelVersion>4.0.0</modelVersion>
     <groupId>com.github.madprogger</groupId>
-    <artifactId>maven-dependency-duplicator-plugin</artifactId>
+    <artifactId>dependency-duplicator-plugin</artifactId>
     <version>0.3.0-SNAPSHOT</version>
 
     <name>dependency duplicator build plugin</name>
     <description>filtered dependency duplicator with configurable changes</description>
-    <url>https://github.com/mickroll/maven-dependency-duplicator-plugin</url>
+    <url>https://github.com/mickroll/dependency-duplicator-plugin</url>
 
     <scm>
-        <url>https://github.com/mickroll/maven-dependency-duplicator-plugin.git</url>
+        <url>https://github.com/mickroll/dependency-duplicator-plugin.git</url>
     </scm>
 
     <issueManagement>
-        <url>https://github.com/mickroll/maven-dependency-duplicator-plugin/issues</url>
+        <url>https://github.com/mickroll/dependency-duplicator-plugin/issues</url>
         <system>GitHub Issues</system>
     </issueManagement>
 

--- a/plugin/src/main/java/com/github/mickroll/maven/dependency_duplicator_plugin/PluginLifecycleParticipant.java
+++ b/plugin/src/main/java/com/github/mickroll/maven/dependency_duplicator_plugin/PluginLifecycleParticipant.java
@@ -66,7 +66,7 @@ public class PluginLifecycleParticipant extends AbstractMavenLifecycleParticipan
 
         if (session.getProjectDependencyGraph() == null) {
             // may happen, if module graph is not valid
-            logger.warn("Execution of maven-dependency-duplicator-plugin is not supported in this environment: "
+            logger.warn("Execution of dependency-duplicator-plugin is not supported in this environment: "
                     + "Current MavenSession does not provide a ProjectDependencyGraph.");
             return;
         }
@@ -85,7 +85,7 @@ public class PluginLifecycleParticipant extends AbstractMavenLifecycleParticipan
         MavenProject projectCursor = project;
         Plugin plugin = null;
         while (projectCursor != null) {
-            plugin = projectCursor.getPlugin("com.github.madprogger:maven-dependency-duplicator-plugin");
+            plugin = projectCursor.getPlugin(PluginMojo.PLUGIN_KEY);
             if (plugin != null) {
                 // break to keep correct projectCursor
                 break;
@@ -103,7 +103,7 @@ public class PluginLifecycleParticipant extends AbstractMavenLifecycleParticipan
         }
 
         try {
-            final MojoDescriptor descriptor = mojoDescriptorCreator.getMojoDescriptor(plugin.getKey() + ":" + PluginMojo.GOAL, session, projectCursor);
+            final MojoDescriptor descriptor = mojoDescriptorCreator.getMojoDescriptor(PluginMojo.PLUGIN_KEY + ":" + PluginMojo.GOAL, session, projectCursor);
             logger.debug("descriptor created " + descriptor);
 
             descriptor.setConfiguration(getConfig(plugin.getConfiguration()));

--- a/plugin/src/main/java/com/github/mickroll/maven/dependency_duplicator_plugin/config/PluginMojo.java
+++ b/plugin/src/main/java/com/github/mickroll/maven/dependency_duplicator_plugin/config/PluginMojo.java
@@ -15,6 +15,8 @@ public class PluginMojo extends AbstractMojo {
 
     public static final String GOAL = "duplicate-dependencies";
 
+    public static final String PLUGIN_KEY = "com.github.madprogger:dependency-duplicator-plugin";
+
     /**
      * Duplications to match and execute against existing dependencies.
      * <p>

--- a/pom.xml
+++ b/pom.xml
@@ -4,20 +4,20 @@
     xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
     <modelVersion>4.0.0</modelVersion>
     <groupId>com.github.madprogger</groupId>
-    <artifactId>maven-dependency-duplicator-plugin-parent</artifactId>
+    <artifactId>dependency-duplicator-plugin-parent</artifactId>
     <version>0.3.0-SNAPSHOT</version>
     <packaging>pom</packaging>
 
     <name>dependency duplicator build plugin - parent</name>
 
-    <url>https://github.com/mickroll/maven-dependency-duplicator-plugin</url>
+    <url>https://github.com/mickroll/dependency-duplicator-plugin</url>
 
     <scm>
-        <url>https://github.com/mickroll/maven-dependency-duplicator-plugin.git</url>
+        <url>https://github.com/mickroll/dependency-duplicator-plugin.git</url>
     </scm>
 
     <issueManagement>
-        <url>https://github.com/mickroll/maven-dependency-duplicator-plugin/issues</url>
+        <url>https://github.com/mickroll/dependency-duplicator-plugin/issues</url>
         <system>GitHub Issues</system>
     </issueManagement>
 

